packages/backend-core/src/redis/redlockImpl.ts

```diff
@@ -105,13 +106,39 @@ export async function doWithLock<T>(
   task: () => Promise<T>
 ): Promise<RedlockExecution<T>> {
   const redlock = await getClient(opts.type, opts.customOptions)
-  let lock
+  let lock: Redlock.Lock | undefined
+  let timeout
   try {
     const name = getLockName(opts)
 
     // create the lock
     lock = await redlock.lock(name, opts.ttl)
 
+    if (opts.type === LockType.AUTO_EXTEND) {
+      // We keep extending the lock while the task is running
+      const extendInIntervals = (): void => {
+        timeout = setTimeout(async () => {
+          let isExpired = false
+          try {
+            lock = await lock!.extend(opts.ttl)
+          } catch (err: any) {
+            isExpired = err.message.includes("Cannot extend lock on resource")
+            if (isExpired) {
+              console.error("The lock expired", { name })
+            } else {
+              throw err
+            }
+          }
+
+          if (!isExpired) {
+            extendInIntervals()
+          }
+        }, opts.ttl / 2)
```

Comment:
Does the user still have the ability to set the TTL? Only worry is if a really short TTL was set, like 100ms, then this will end up trying to extend every 50ms, which likely won't be possible/could lead to failures to extend - maybe we should have a minimum TTL for auto-extend?